consensus/src/rand/rand_gen/rand_manager.rs

```diff
@@ -178,28 +178,40 @@ impl<S: Share, D: AugmentedData, Storage: AugDataStorage<D>> RandManager<S, D, S
         }
     }
 
-    fn aggregate_shares_task(&self, metadata: RandMetadata) -> DropGuard {
+    fn spawn_aggregate_shares_task(&self, metadata: RandMetadata) -> DropGuard {
         let rb = self.reliable_broadcast.clone();
         let aggregate_state = Arc::new(ShareAggregateState::new(
             self.rand_store.clone(),
             metadata.clone(),
             self.config.clone(),
         ));
-        let epoch = metadata.epoch();
+        let epoch_state = self.epoch_state.clone();
         let round = metadata.round();
-        let request = RequestShare::new(self.epoch_state.epoch, metadata);
+        let rand_store = self.rand_store.clone();
         let task = async move {
-            info!(
-                epoch = epoch,
-                round = round,
-                "[RandManager] Start broadcasting share request",
-            );
-            rb.broadcast(request, aggregate_state).await;
-            info!(
-                epoch = epoch,
-                round = round,
-                "[RandManager] Finish broadcasting share request",
-            );
+            tokio::time::sleep(Duration::from_millis(300)).await;
+            let maybe_existing_shares = rand_store.lock().get_all_shares_authors(&metadata);
+            if let Some(existing_shares) = maybe_existing_shares {
```

Comment:
If existing shares are sufficient for generating decision, skip this?

Reply:
if existing is enough/already decided, it returns None